src/averageAges.js

```diff
@@ -37,7 +43,16 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const women = withChildren
+    ? people.filter((wom) => people.some((p) => wom.name === p.mother))
+    : people.filter((wom) => wom.sex === 'f');
+
+  const sumAge = women.reduce((prev, current) => prev
+  + (current.died - current.born), 0);
+
+  const average = sumAge / women.length;
```

Comment:
DRY
Repeatable code move to common helper functions and reuse it. Fix all cases please